testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/KeycloakQuarkusServerDeployableContainer.java

```diff
@@ -142,17 +163,35 @@ private Process startContainer() throws IOException {
         builder.environment().put("KEYCLOAK_ADMIN_PASSWORD", "admin");
 
         if (restart.compareAndSet(false, true)) {
-            FileUtils.deleteDirectory(configuration.getProvidersPath().resolve("data").toFile());
+            if (SystemUtils.IS_OS_WINDOWS) {
+                deleteTempFilesOnWindows(configuration.getProvidersPath().resolve("data"));
+            } else {
+                FileUtils.deleteDirectory(configuration.getProvidersPath().resolve("data").toFile());
+            }
         }
 
         return builder.start();
     }
 
     private String[] getProcessCommands() {
         List<String> commands = new ArrayList<>();
-        commands.add(getCommand());
-        commands.add("-v");
+
+        if(SystemUtils.IS_OS_WINDOWS) {
+            commands.add(configuration.getProvidersPath().resolve("bin") + File.separator + SCRIPT_CMD_INVOKABLE);
+        } else {
+            commands.add(SCRIPT_CMD_INVOKABLE);
+            commands.add("-v"); // see #11185, has to be fixed before using -v on windows.
+        }
+
         commands.add("start");
+
+        if (SystemUtils.IS_OS_WINDOWS && firstStart.compareAndSet(true, false)) {
```

Comment:
We have a similar logic already for server restarts or when running cluster tests. Also not sure if you need `firstStart` if there is a `restart` flag.

Reply:
`firstStart` is effectively avoiding the ant task, mimicking the behaviour of the `[configure.xml](https://github.com/keycloak/keycloak/blob/240de2aeb0fef64b050838aa3c3dfb9ef380aa2b/testsuite/integration-arquillian/servers/auth-server/quarkus/ant/configure.xml)` we use for unix, to avoid setting the (dynamic) full-path to jar as outlined in the description (see also: https://ant.apache.org/manual/Tasks/exec.html ).  We could think about re-using the `restart` flag for it, but in my head it made more sense at nearly no cost to add a specific flag. Not hard on this, though.

Reply:
Let's keep it simple and re-use if possible.